### PR TITLE
feat: add erc4626 custom vault

### PR DESCRIPTION
**Pull request by pakim249CAL**
#98

### CODE REVIEW

File: src/ERC4626UpgradeableSafe.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /* CONSTRUCTOR */
+
+    /// @dev The contract automatically disables initializers when deployed so that nobody can highjack the implementation contract.
+    constructor() {
+        _disableInitializers();
+    }
+
+    /* INITIALIZER */
+
+    function __ERC4626UpgradeableSafe_init(IERC20MetadataUpgradeable asset, uint256 initialDeposit)
```
Comment (MerlinEgalite) [inline src/ERC4626UpgradeableSafe.sol:24]: To be honest I think that `IERC20MetadataUpgradeable` is more annoying the anything, wdyt?

File: remappings.txt [hunk 2]
```diff
@@ -2,3 +2,4 @@
 ds-test/=lib/forge-std/lib/ds-test/src/
 forge-std/=lib/forge-std/src/
 openzeppelin-contracts/=lib/openzeppelin-contracts/
+openzeppelin-upgradeable/=lib/openzeppelin-contracts-upgradeable/
```
Comment (Rubilmax) [inline remappings.txt:5]:
```suggestion
openzeppelin-contracts/=lib/openzeppelin-contracts/contracts/
openzeppelin-upgradeable/=lib/openzeppelin-contracts-upgradeable/contracts/
```

**Review by Rubilmax — DISMISSED** [review]

File: src/ERC4626UpgradeableSafe.sol (hunk 1 above)
Comment (MerlinEgalite) [reply src/ERC4626UpgradeableSafe.sol:24]: Yes but let's cast to a more commomn ERC20 interface and not IERC20MetadataUpgradeable which is not commonly used. This one is annoying to use

Comment (MerlinEgalite) [reply src/ERC4626UpgradeableSafe.sol:24]: But anyway we can keep this one not a blocker

Comment (Rubilmax) [reply src/ERC4626UpgradeableSafe.sol:24]: Agreed in this case, but there are other concerns: OZ may force us to use this interface
Btw, it seems the OZ dependency is not up-to-date

Comment (MerlinEgalite) [reply src/ERC4626UpgradeableSafe.sol:24]: For [v4.8.3](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/releases/tag/v4.8.3)

Knowing that the 4626 implement has been modified in v4.8.0 and v4.8.1 must everything seems ok. I suggest to move to the v4.8.3 (last release)

**Review by MerlinEgalite — DISMISSED** [review]

**Review by Tristan22400 — DISMISSED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by MathisGD — DISMISSED** [review]

**Comment by MerlinEgalite** [comment]
go meeeerge

**Review by MerlinEgalite — APPROVED** [review]

**Review by MathisGD — APPROVED** [review]

**Review by Tristan22400 — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Pull request merged**